### PR TITLE
refactor(registry): 更改注册表结构，覆盖配置移至overwrite字段下

### PR DESCRIPTION
**Pull request by gene9831**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

### What is the new behavior?

`template`, `methods`, `lifeCycle` 可以覆盖的配置移动至overwrite字段下

```js
{
  id: 'engine.type.id',
  // 覆盖逻辑
  overwrite: {
    template: TEMLATE,
    methods: {
      '': {
        handleClick: (ctx) => () => {}
      }
    },
    lifeCycle: {
      '': {
        onMounted: [
          (ctx) => () => {
            console.log('onMounted 1')
          },
          (ctx) => () => {
            console.log('onMounted 2')
          }
        ]
      }
    }
  }
}
```

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
  - Enhanced registry structure for better organization, including new properties for layout, toolbars, settings, canvas, and utilities.

- **Improvements**
  - Improved handling of templates, life cycles, and methods to support more flexible configurations with the new `overwrite` property.

- **Bug Fixes**
  - Refined internal logic to ensure more accurate and efficient data processing.

### CODE REVIEW

File: packages/entry/src/common.js [hunk 1]
```diff
@@ -88,20 +88,25 @@ const handleLifeCycles = (id, lifeCycles) => {
...
+  const { layout, overwrite } = value
+
   if (layout) {
     layoutHashMap[id] = layout
   }
-  if (methods) {
-    handleMethods(id, methods)
+
+  if (overwrite) {
+    const { template, lifeCycles, methods } = overwrite
```
Comment (chilingling) [inline packages/entry/src/common.js:117]: overwrite 这里感觉是不是可以做一下容错判断，防止用户传进来的 `overwrite` 不是 plainobject

File: packages/entry/src/common.js [hunk 2]
```diff
@@ -107,20 +107,25 @@ const handleLifeCycles = (id, lifeCycles) => {
...
+    if (template) {
+      templateHashMap[id] = template
+    }
+    // 处理生命周期
+    if (lifeCycles) {
+      handleLifeCycles(id, lifeCycles)
+    }
+    if (methods) {
+      handleMethods(id, methods)
+    }
```
[the remark this answers is not in the document]
Comment (gene9831) [reply packages/entry/src/common.js]: udpated

File: packages/entry/src/common.js [hunk 3]
```diff
@@ -88,20 +88,25 @@ const handleLifeCycles = (id, lifeCycles) => {
...
```
[the remark this answers is not in the document]
Comment (gene9831) [reply packages/entry/src/common.js]: updated

File: packages/entry/src/common.js (hunk 1 above)
Comment (gene9831) [reply packages/entry/src/common.js:117]: updated. `typeof overwrite === 'object' && overwrite`

**Review by chilingling — APPROVED** [review]

**Review by hexqi — APPROVED** [review]

**Pull request merged**